examples/NAT.cs

```diff
@@ -36,223 +36,348 @@
 using PacketDotNet;
 using System.Net;
 using Pax;
+using System.Net.NetworkInformation;
 
 public class NAT : SimplePacketProcessor {
   public const int Port_Drop = -1;
   public const int Port_Outside = 0;
 
   private IPAddress my_address;
   private ushort next_port;
+  private readonly object portLock = new object();
```

Comment:
IIRC you mentioned getting weird results when a different object (was it `next_port`) was used for locking. Did you form an idea why this was the case?